packages/react-search-ui/src/containers/BeaconFacet.tsx

```diff
@@ -0,0 +1,263 @@
+  children view. This is different from the original Facet component as it only passes 
+  facet values for the single filter type specified as a parameter to the children view. The access
+  to facet values for the "none" filter type allows the children view to display negatable 
+  facet values with the correct selected state in addition to correctly displaying positive 
+  facet values. Moreover, the component passes several filter state handlers to the children view 
+  so that the children view can handle state updates to both positive and negatable facet values.
+  (See `addFilter`, `removeFilter`, and `setFilter` in the code below.)
+
+  2) Default Options - The component accepts default options as an optional parameter which can
+  be used to display certain facet values. This is useful for curating or highlighting certain facet
+  values for better use experience. The component can also be configured to either display or hide
+  other facet values that are not in the default options list. If the component is configured to
+  display other facet values, it will make sure to prevent displaying duplicate facet values that
+  have already appeared in the default options list.
+
+# Parameters
+Only listing custom parameters introduced in addition to the ones from the original Facet component.
+For the list of original parameters, see the original Facet component.
+- @param {FacetValue[]} defaultOptions - The list of default options to display. If specified, the
+  component will display the default options in the order specified in the lis
+- @param {boolean} showDefaultOptionsOnly - Whether to display only the default options or not. If
+  true, it will hide other facet values that are not in the default options list.
+
+# Implementation Notes
+The foundation of this component is the original Facet component provided by ElasticSearch and the
+code is copied and pasted from that component as a starting point. Then, custom code was written
+to implement the features described above. To keep up with the changes made to the original Facet
+component in future releases, each release should be carefully reviewed and any corresponding
+changes should also be made to this component.  
+*/
+
+import React from "react";
+import { Component } from "react";
+import {
+  FacetContainerContext,
+  BeaconFacetContainerProps,
+  BeaconFacetViewProps,
+  MultiCheckboxFacet,
+  BeaconFacetValuesMapProps
+} from "@elastic/react-search-ui-views";
+import { FacetValue, helpers } from "@elastic/search-ui";
+import {
+  isMatchedFacetValueAndSearchTerm,
+  isMatchedDefaultOptionAndSearchTerm,
+  buildFacetValuesMap,
+  stringifyFilterValue
+} from "../helpers";
+import { withSearch } from "..";
+
+const { markSelectedFacetValuesFromFilters } = helpers;
+
+type FacetContainerState = {
+  searchTerm: string;
+  more: number;
+};
+
+export class BeaconFacetContainer extends Component<
```

Comment:
Why opt to make this a class component over a function component? Is it because we need to implement some interface that the normal `FacetContainer` extends?

Reply:
I also had a preference for functional component but this was still written as a class component for convenience.

The original `Facet` component written by Elastic team is still written as a class component so my starting point was to duplicate that file and make edits.
https://github.com/Tsquare/search-ui/blob/master/packages/react-search-ui/src/containers/Facet.js.

Moreover, I thought keeping this component as a class component would make it slightly easier to track and apply any future changes that the Elastic makes to their original component